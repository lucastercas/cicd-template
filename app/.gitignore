target
bin
.env
.gradle
.mvn