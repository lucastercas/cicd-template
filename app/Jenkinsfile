pipeline {
    agent any
    environment {
        PROJECT_KEY = 'org.springframework:gs-spring-boot-docker'
        SONAR_LOGIN = 'e6f696687d9b343872358aaf19371fe5dfdca63f'
        SONAR_URL = 'http://sonarqube:9000'
    }
    options {
        skipStagesAfterUnstable()
    }
    stages {
        stage('Build Package') {
            steps {
                sh "make build"
            }
        }
        stage('Test') {
            steps {
                sh 'make test'
            }
            post {
                always {
                    junit 'target/surefire-reports/*.xml'
                }
            }
        }
        stage('Sonarqube') {
            steps {
                sh "mvn sonar:sonar -Dsonar.projectKey=${PROJECT_KEY} -Dsonar.host.url=${SONAR_URL} -Dsonar.login=${SONAR_LOGIN}"
            }
        }
        stage('Docker Login') {
            steps {
                sh "docker login -u ${DOCKER_USER} -p ${DOCKER_PASSWORD}"
            }
        }
        stage('Build Docker Image') {
            steps {
                sh 'make build_image'
            }
        }
        stage('Push Image') {
            steps {
                sh 'make push_image'
            }
        }
    }
    post {
        always {
            archiveArtifacts artifacts: "target/*.jar", fingerprint: true
        }
    }
}