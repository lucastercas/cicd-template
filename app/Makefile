.PHONY: build_jar test scan build_image push_image

build:
	@docker run --rm \
		-v "$(CURDIR)":/usr/src/mymaven \
		-v /root/.m2:/root/.m2 \
		-w /usr/src/mymaven \
		maven:latest mvn -B -DskipTests clean package

test:
	@docker run --rm \
		-v "$(CURDIR)":/usr/src/mymaven \
		-v /root/.m2:/root/.m2 \
		-w /usr/src/mymaven \
		maven:latest mvn test

scan:
	@mvn sonar:sonar \
			-Dsonar.projectKey=org.springframework:gs-spring-boot-docker \
			-Dsonar.host.url=http://$(SONAR_HOST) \
			-Dsonar.login=$(SONAR_LOGIN)

build_image:
	@docker build -t lucastercas/spring .

push_image:
	@docker push lucastercas/spring