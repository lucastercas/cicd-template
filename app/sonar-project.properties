sonar.projectKey=
sonar.sources=.
sonar.exclusions=framework/pb/*
sonar.test.exclusions=framework/pb/*
