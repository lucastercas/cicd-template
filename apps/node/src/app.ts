import Koa from "koa";
import Router from "koa-router";

// import * as logger from "koa-logger";
// import * as json from "koa-json";

const app = new Koa();
const router = new Router();

router.get("/", (ctx: Koa.context) => {
  ctx.body = "Hello There!";
  // await next();
});

// app.use(json());
// app.use(logger());

app.use(router.routes());
// app.use(router.routes()).use(router.allowedMethods());

export = app;
