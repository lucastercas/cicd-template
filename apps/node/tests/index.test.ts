import request from "supertest";
import app from "../src/app";

test("Hello work works", async () => {
  const response = await request(app.callback()).get("/");
  expect(response.status).toBe(200);
  expect(response.text).toMatchSnapshot();
});
