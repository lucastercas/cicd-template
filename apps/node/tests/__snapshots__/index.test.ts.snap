// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello work works 1`] = `"Hello There!"`;
